network-science-lab/top-spreaders-dataset
# Feature  oop generation

I've modified @adampirog 's proposal so that it can handle more models and it meets our requirements better. Here are main changes:
- I've implemented a Preferential Attachment model (which can be considered as a sibling of the Barabasi-Albert model)
- I've designed an abstraction over PA and ER models following OOP paradigm
- I've adjusted CLI configuration so that we only have to provide: number of networks to generate, min and max numbers of actors and layers, network model type to harness
- ER model selects number of nodes in each layer following Gaussian distribution with: expected value equals number of actors and with standard deviation equals 10% of the former
- PA model picks `m0` (aka number of hubs) and `m`(aka number of links to attach with for each new node) as follows: `m0` is 5% of given number of actors (but not less than 3) and `m=`m0`
- since fixing RNG didn't result for me in reproducible results, I've removed it


To test the code please run:
- `nsl_data_utils/misc/generators.py` to see degree distributions of one network generated with each model
- `nsl_data_utils/misc/generatore.py` and uncomment lines 71-81 to see a demo of the main routine
